financial-services-accelerator/consent-mgt-api/org.wso2.financial.services.accelerator.consent.mgt.api.service/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/api/service/impl/ConsentCoreServiceImpl.java

```diff
@@ -0,0 +1,852 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.financial.services.accelerator.consent.mgt.api.service.impl;
+
+import net.minidev.json.JSONObject;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.ConsentCoreDAO;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.constants.ConsentError;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.exceptions.ConsentDataDeletionException;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.exceptions.ConsentDataInsertionException;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.exceptions.ConsentDataRetrievalException;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.exceptions.ConsentDataUpdationException;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.exceptions.ConsentMgtException;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.models.AuthorizationResource;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.models.ConsentAttributes;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.models.ConsentHistoryResource;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.models.ConsentResource;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.models.ConsentStatusAuditRecord;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.models.DetailedConsentResource;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.persistence.ConsentStoreInitializer;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.util.DatabaseUtils;
+import org.wso2.financial.services.accelerator.consent.mgt.api.service.ConsentCoreService;
+import org.wso2.financial.services.accelerator.consent.mgt.api.service.constants.ConsentCoreServiceConstants;
+import org.wso2.financial.services.accelerator.consent.mgt.api.service.util.ConsentCoreServiceUtil;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * Consent core service implementation.
+ */
+public class ConsentCoreServiceImpl implements ConsentCoreService {
+
+    private static final Log log = LogFactory.getLog(ConsentCoreServiceImpl.class);
+
+    private static ConsentCoreServiceImpl consentCoreService = null;
+
+    public static ConsentCoreServiceImpl getInstance() {
+        if (consentCoreService == null) {
+            consentCoreService = new ConsentCoreServiceImpl();
+        }
+        return consentCoreService;
+    }
```

Comment:
invalid usage of singleton pattern. Please check other places as well.

```suggestion
    private static volatile ConsentCoreServiceImpl instance;
    private final ConsentCoreDAO consentCoreDao;
    
    public static ConsentCoreServiceImpl getInstance() {
        if (instance == null) {
            synchronized (ConsentCoreServiceImpl.class) {
                if (instance == null) {
                    instance = new ConsentCoreServiceImpl();
                }
            }
        }
        return instance;
    }
    
    private ConsentCoreServiceImpl() {
        this.consentCoreDao = ConsentStoreInitializer.getInitializedConsentCoreDAOImpl();
    }
```

Reply:
resolved in https://github.com/wso2/financial-services-accelerator/commit/e4dcb09e4d33a0969d5ff07658ff119ef6fa5e9c